### ISSUE TITLE
Might be possible to make functions used in setup(), i.e., ```find_all()```, ```get_metadata()```, ```get_blockdevs()``` more efficient as a class

### ISSUE THREAD

**Issue by mulkieran** [stratis-storage/stratisd#486]
~~Probably ```find_all()``` should go to engine and the other two to pool. There are probably ways that they could be simplified.~~

~~This is not a task that should be attempted in the very near future.~~

As a result of cache tier work, these methods have been distributed and altered in a variety of ways. But they have also been made less public. This makes it more likely that they can be called only in contexts where more restrictive preconditions can be met. So this is a goal that is still worth working for.

**Comment by agrover** [comment]
This implicitly includes the excessive opens/reads mentioned in #661.

**Comment by mulkieran** [comment]
In PR #837 :

```find_all``` is used exclusively in ```StratEngine::initialize```.
```get_blockdevs``` is use exclusively in ```Backstore::setup```.
```get_metadata``` is used exclusively in ```setup::setup_pool```.
```is_stratis_device``` is used in ```setup::find_all``` and ```StratEngine::block_allocate```
```setup_pool``` is used in ```StratEngine``` in ```initialize``` and ```block_allocate```.

**Comment by mulkieran** [comment]
blocked by libblkid work.

**Comment by mulkieran** [comment]
These methods were completely reworked to support encryption, and are about as efficient as can be managed at this time. For example, previously the BDA was read multiple times, now it is always just read and stored once on any attempt to set up a pool. Closing.
